src/cider/nrepl/middleware/out.clj

```diff
@@ -0,0 +1,86 @@
+(ns cider.nrepl.middleware.out
+  "Change *out* to print on connections in addition to process out.
+
+  Automatically changes the root binding of *out* to print to any
+  active connections. An active connection is a session that has sent
+  at least one \"eval\" op.
+
+  We use an eval message, instead of the clone op, because there's no
+  guarantee that the channel that sent the clone message will properly
+  handle output replies."
+  (:require [cider.nrepl.middleware.util.cljs :as cljs]
+            [clojure.tools.nrepl.middleware :refer [set-descriptor!]]
+            [clojure.tools.nrepl.middleware.interruptible-eval :as ie]
+            [clojure.tools.nrepl.middleware.session :as session])
+  (:import [java.io PrintWriter Writer]))
+
+;;; OutStream
+(defonce original-out *out*)
+
+(declare eval-messages)
+
+(defmacro do-out-messages
+  "Run body with v bound to the output stream of each msg in msg-seq.
```

Comment:
It's customary to use the `with-` prefix for macros which set up new execution contexts.